docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,192 @@
+# Jet job member pruning
+
+|||
+|---|---|
+|Related Jira|[HZ-1605](https://hazelcast.atlassian.net/browse/HZ-1605)|
+|Document Status / Completeness|IN PROGRESS|
+|Requirement owner|Sandeep Akhouri|
+|Developer(s)|Sasha Syrotenko|
+|Quality Engineer|Isaac Sumner|
+|Technical Reviewers|TBD|
+|Version|5.4|
+
+## Background
+
+Before Hazelcast Platform 5.4, Jet job was always deployed to all data members, specifically all non-lite members. 
+If some DAG vertex isn’t using all members, it creates no-op processors on the rest, but Jet still creates queues 
+to/from those vertices and starts the processors, even though it completes immediately and the queues are closed with a `DONE_ITEM`. 
+If some member isn’t used at all, the DAG is still deployed to it, and the coordinator has to send `InitExecutionOperation` to it and wait for 
+the completion. Even though the processors are no-op, or have no data to process, it’s an unnecessary overhead, 
+which becomes noticeable in very small batch jobs and/or large clusters due to serial transmission.
```

Comment:
what does it mean "serial transmission"? InitExecutionOperation is sent in parallel to all members, isn't it?

Reply:
Removed